Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -2166,8 +2182,7 @@ static bool isInRecoveryOS()
     clearAllEditCommands();
 
     if (!m_isPreparingToUnparentView) {
-        NSWindow *stopObservingWindow = m_targetWindowForMovePreparation.get() ?: [m_view window];
-        [m_windowVisibilityObserver stopObserving:stopObservingWindow];
+        [m_windowVisibilityObserver stopObserving];
         [m_windowVisibilityObserver startObserving:window];
         m_isObservingWindow = true;
```

Comment:
Maybe we should move `m_isObservingWindow` into WKWindowVisibilityObserver, then check it in `startObserving` and `stopObserving`. And we could  assert `!m_isObservingWindow` in dealloc?